### PR TITLE
Dart: install executables into $HOME, not the working directory

### PR DESCRIPTION
**Pull request by kevmoo**
This help keep the working directory clean, which is great for tools that are looking for Dart sources, etc.

CC @a14n @devoncarew @sethladd

[Review on Reviewable](https://reviewable.io/reviews/travis-ci/travis-build/463)

### CODE REVIEW

**Comment by kevmoo** [comment]
@BanzaiMan do I bother you about this?

**Comment by a14n** [comment]
Looks good to me.

**Comment by sethladd** [comment]
Thanks!

File: lib/travis/build/script/dart.rb [hunk 1]
```diff
@@ -102,6 +102,10 @@ def script
 
         private
 
+          def download_bin(url, foo)
```
Comment (BanzaiMan) [inline lib/travis/build/script/dart.rb]: What is this method for?

Comment (kevmoo) [inline lib/travis/build/script/dart.rb]: A forgotten idea. Will fix.

Comment (kevmoo) [inline lib/travis/build/script/dart.rb]: @BanzaiMan See updated pull request

**Comment by BanzaiMan** [comment]
Thanks! Deployed.

**Comment by kevmoo** [comment]
Validated the output. Thanks, @BanzaiMan

**Pull request merged**